.github/workflows/nix.yml

```diff
@@ -0,0 +1,20 @@
+name: Nix derivation checks
+
+on:
+  push:
+    branches: ["main"]
```

Comment:
do you want to update the nix hash on every push to main, or when we do releases?

Reply:
I think it's only pushes to main where mix.lock changed.

Reply:
yes, but do we want all pushes to main that change the mix lock to update the nix flake, or should the flake get changed only when we release?

Reply:
I think it should be any change on main to mix.lock, so that main is always usable for nix users.

Reply:
i'm unfamiliar with how nix works, so ok.

Reply:
My point earlier was just that I don't think it will be very noisy, since mix.lock doesn't change often.

Reply:
@scohen conceptually it is pretty easy - you point to the Git ref in this repo, it fetches project and build it. The thing is that it checks the dependencies hash against the fetched deps. So this simply makes it in a way that will allow to pin any (updated) commit after `mix.lock` was updated.